rust/worker/src/compactor/compaction_manager.rs

```diff
@@ -293,25 +295,42 @@ impl Debug for CompactionManager {
 
 // ============== Handlers ==============
 #[async_trait]
-impl Handler<ScheduleMessage> for CompactionManager {
+impl Handler<ScheduledCompactionMessage> for CompactionManager {
     type Result = ();
 
     async fn handle(
         &mut self,
-        _message: ScheduleMessage,
+        _message: ScheduledCompactionMessage,
         ctx: &ComponentContext<CompactionManager>,
     ) {
-        println!("CompactionManager: Performing compaction");
+        tracing::info!("CompactionManager: Performing compaction");
         let mut ids = Vec::new();
         self.compact_batch(&mut ids).await;
 
         self.hnsw_index_provider.purge_by_id(&ids).await;
 
         // Compaction is done, schedule the next compaction
-        ctx.scheduler
-            .schedule(ScheduleMessage {}, self.compaction_interval, ctx, || {
-                Some(span!(parent: None, tracing::Level::INFO, "Scheduled compaction"))
-            });
+        ctx.scheduler.schedule(
+            ScheduledCompactionMessage {},
+            self.compaction_interval,
+            ctx,
+            || Some(span!(parent: None, tracing::Level::INFO, "Scheduled compaction")),
+        );
+    }
+}
+
+#[async_trait]
+impl Handler<OneOffCompactionMessage> for CompactionManager {
+    type Result = ();
+    async fn handle(
+        &mut self,
+        _message: OneOffCompactionMessage,
+        _ctx: &ComponentContext<CompactionManager>,
+    ) {
+        tracing::info!("CompactionManager: Performing compaction");
```

Comment:
I think that this should not synchronously perform compaction but instead queue the id for compaction in the next scheduled run